graylog2-server/src/main/java/org/graylog/plugins/views/search/Query.java

```diff
@@ -242,4 +247,36 @@ public Query build() {
             return autoBuild();
         }
     }
+
+    // This code assumes that we only add streams via gui shallow.
```

Comment:
I'd make this a TODO:
Suggestion: 
// TODO This code assumes that we only use shallow filters for streams. If this ever changes, we need to implement a mapper that can handle filter trees.